### PR TITLE
[DOCS] add DNS discovery setup guide

### PR DESCRIPTION
**Pull request by fjl**
This adds setup instructions for the DHT crawler and DNS node list publishing tools.

### CODE REVIEW

File: docs/_developers/dns-discovery-setup.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+title: DNS Discovery Setup Guide
+sort_key: C
+---
+
+This document explains how to set up an [EIP 1459][dns-eip] node list using the cmd/devp2p
```
Comment (adamschmideg) [inline docs/_developers/dns-discovery-setup.md]: Can you link to the tool? Do you mean https://github.com/ethereum/go-ethereum/tree/master/cmd/p2psim? Or is it on another branch?

File: docs/_developers/dns-discovery-setup.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+and public testnets, but you may also find this helpful if you want to set up DNS-based
+discovery for a private network.
+
+DNS-based node lists can serve as a fallback option when connectivity to the discovery DHT
+is unavailable. In this guide, we'll create node lists by crawling the discovery DHT, then
+publishing the resulting node sets under chosen DNS names.
+
+### Installing the devp2p command
+
+cmd/devp2p is a developer utility and is not included in the Geth distribution. You can
```
Comment (adamschmideg) [inline docs/_developers/dns-discovery-setup.md]: Oh, I see, it's not part of Geth. Why? I mean you may want to write a sentence on the reasoning behind it.

File: docs/_developers/dns-discovery-setup.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+```shell
+devp2p nodeset filter all-nodes.json -eth-network mainnet > mainnet.nodes.example.org/nodes.json
+```
+
+The following filter flags are available:
+
+* `-eth-network ( mainnet | ropsten | rinkeby | goerli )` selects an Ethereum network.
+* `-les-server` selects LES server nodes.
+* `-ip <mask>` restricts nodes to the given IP range.
+* `-min-age <duration>` restricts the result set to nodes which have been live for the
```
Comment (ligi) [inline docs/_developers/dns-discovery-setup.md]: perhaps state the unit here - like "in seconds"
I also would remove the "set" here - makes it easier to read

File: docs/_developers/dns-discovery-setup.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  given duration.
+
+### Creating DNS trees
+
+To turn a node list into a DNS node tree, the list needs to be signed. To do this, you
+need a key pair. To create the key file in the right format, you can use the cmd/ethkey
```
Comment (ligi) [inline docs/_developers/dns-discovery-setup.md]: suggestion: replace right with correct

File: docs/_developers/dns-discovery-setup.md (hunk 2 above)
Comment (fjl) [reply docs/_developers/dns-discovery-setup.md]: I don't want to explain why cmd/devp2p functionality is not in geth in this document. It's a developer tool for peer-to-peer networking tasks.

File: docs/_developers/dns-discovery-setup.md (hunk 1 above)
Comment (fjl) [reply docs/_developers/dns-discovery-setup.md]: The link to the EIP is right there. I will add more info to this document once we have a way to actually use DNS discovery in geth.

**Pull request merged**